File: AlgorithmRise/extra/LeetCode/剑指offer-LCOF/面试题32 - I. 从上到下打印二叉树/wrong-answer/wrong-answer-1.go
package wronganswer

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 var nodes map[int][]int

 func traverseTree(node *TreeNode, level int) int {
 
	 if nil == node {
		 return 0
	 }
 
	nodes[level] = append(nodes[level], node.Val)
 
	 if nil != node.Left {
		 traverseTree(node.Left, level+1)
	 }
 
	 if nil != node.Right {
		 traverseTree(node.Right, level+1)
	 }
 
	 return 0
 }
 
 func levelOrder(root *TreeNode) []int {
 
	 nodes = make(map[int][]int)
 
	 // 定义一个二维数组
	 traverseTree(root, 0)
 
	 t := make([]int, 0)
 
	 for _, arr := range nodes{
 
		 for _, val := range arr{
			 t = append(t, val)
		 }
	 }
 
	 fmt.Println(nodes)
 
	 return t
 }

File: AlgorithmRise/extra/LeetCode/Algorithms/112. Path Sum/wrong-answer-1.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool FLAG = false;
int SUM = 0;
int traverseTree(struct TreeNode* p, int sum){

    if(!p){
        return 0;
    }

    sum += p->val;

    if(p->left){
        traverseTree(p->left, sum);
    }
    if(p->right){
        traverseTree(p->right, sum);
    }

    if(SUM == sum){
        FLAG = true;
    }

    return sum;
}

bool hasPathSum(struct TreeNode* root, int sum){

    SUM = sum;
    FLAG = false;
    if(!root){
        return 0 == sum;
    }

    traverseTree(root, 0);
    return FLAG;
}